backend/scripts/massage_csv.py

```diff
@@ -0,0 +1,250 @@
+            if randint(1,100) <= weights.get('secondary', 50):
+                new_item['Secondary Name'] = guest_names[randint(0, len(guest_names) - 1)]
+        else:
+            new_item['Placed By'] = 'Roombaht'
+
+
+        if randint(1, 100) <= weights.get('art', 5):
+            new_item['Art Room'] = 'Yes'
+            new_item['Art Name / Placed Name'] = phrasing()
+            new_item['Art Room Type'] = art_type()
+        else:
+            new_item['Art Room'] = 'No'
+
+        output_items.append(new_item)
+
+    return output_items
+
+def massage_guests(input_items):
+    output_items = []
+    name_maps = {}
+    email_maps = {}
+
+    for item in input_items:
+        a_name = "%s %s" % (item['first_name'], item['last_name'])
+        new_name = generate_name(style = 'capital')
+        new_name_bits = new_name.split(' ')
+        new_first_name = new_name_bits[0]
+        new_last_name = new_name_bits[1]
+        new_email = "%s.%s@noop.com" % (new_first_name.lower(), new_last_name.lower())
+
+        if a_name not in name_maps:
+            name_maps[a_name] = "%s %s" % (new_first_name, new_last_name)
+
+        if new_email not in email_maps:
+            email_maps[item['email']] = new_email
+
+
+    # actually anonymize
+    for item in input_items:
+        a_name = "%s %s" % (item['first_name'], item['last_name'])
+        new_first_name, new_last_name = name_maps[a_name].split(' ')
+        new_item = {
+            'invitation_code': item['invitation_code'],
+            'ticket_code': item['ticket_code'],
+            'last_name': new_last_name,
+            'first_name': new_first_name,
+            'email': new_email,
+            'phone': "", #anon
+            'type': item['type'],
+            'product': item['product'],
+            'purchase_date': item['purchase_date'],
+            'ticket_status': item['ticket_status']
+        }
+
+        if item['transferred_from'] != '':
+            new_item['transferred_from'] = name_maps[a_name]
+            new_item['transferred_from_email'] = email_maps[item['email']]
+            new_item['transferred_from_code'] = item['transferred_from_code']
+
+        if item['transferred_to'] != '':
+            new_item['transferred_to'] = name_maps[a_name]
+            new_item['transferred_to_email'] = email_maps[item['email']]
+
+        output_items.append(new_item)
+
+    return output_items
+
+def ingest(filename):
+    if not os.path.exists(filename):
+        raise Exception("input file %s not found" % filename)
+
+    input_dict = []
+    input_items = []
+    with open(filename, "r") as input_handle:
+        input_dict = DictReader(input_handle, skipinitialspace=True)
+        input_fields = [k.lstrip().rstrip() for k in input_dict.fieldnames if type(k)==str]
+        for elem in input_dict:
+            strip_elem = {k.lstrip().rstrip(): v.lstrip().rstrip() for k, v in elem.items() if type(k)==str and type(v)==str}
+            input_items.append(strip_elem)
```

Comment:
🤣 yea maybe we put this DictReader/Writer space stripper stuffs in a module. Seems popular.

Reply:
This has been done and it works on the form uploads too.